backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  Organization,
+  Domain,
+  Vulnerability,
+  connectToDatabase
+} from '../models';
+import { CommandOptions } from './ecs-client';
+import got from 'got';
+import { plainToClass } from 'class-transformer';
+import saveVulnerabilitiesToDb from './helpers/saveVulnerabilitiesToDb';
+import { exit } from 'process';
+import saveDomainsToDb from './helpers/saveDomainsToDb';
+
+interface LGCollectionsResponse {
+  children: any[];
+  createdAt: string;
+  createdBy: string;
+  id: string;
+  name: string;
+  ticScore: number;
+  updatedAt: string;
+  updatedBy: string;
+}
+
+interface LGThreatResponse {
+  totalHits: number;
+  results: {
+    firstSeen: number;
+    lastSeen: number;
+    sources: string[];
+    right: {
+      ticClassificationScores: number[];
+      threatId: string;
+      classifications: string[];
+      ticScore: number;
+      ticCriticality: number;
+      ticSourceScore: number;
+      ticObsInfluence: number;
+      ref: {
+        type: string;
+        id: string;
+      };
+      threatCategories: string[];
+      name: string;
+      type: string;
+      threatName: string;
+    };
+    left: {
+      collectionIds: string[];
+      classifications: string[];
+      ticScore: number;
+      owners: string[];
+      ref: {
+        type: string;
+        id: number;
+      };
+      name: string;
+      type: string;
+      threatNames: string[];
+      locations: {
+        city: string;
+        country: string;
+        region: string;
+        geoPoint: {
+          long: number;
+          lat: number;
+        };
+        countryName: string;
+        sources: string[];
+        country2Digit: string;
+        lastSeen: number;
+      }[];
+      ipv4: number;
+      asns: number[];
+      threatIds: string[];
+      cidrv4s: string[];
+    };
+    ref: {
+      type: string;
+      right: {
+        type: string;
+        id: string;
+      };
+      left: {
+        type: string;
+        id: string;
+      };
+    };
+  }[];
+}
+
+async function getCollectionForCurrentWorkspace() {
+  const resource =
+    'https://delta.lookingglasscyber.com/api/v1/workspaces/' +
+    process.env.LG_WORKSPACE_NAME +
+    '/collections';
+  return (await got(resource, {
+    headers: { Authorization: 'Bearer ' + process.env.LG_API_KEY }
+  }).json()) as LGCollectionsResponse[];
+}
+
+function validateIPAddress(ipAddress) {
+  return /\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(ipAddress);
+}
+
+async function getThreatInfo(collectionID) {
+  const resource = 'https://delta.lookingglasscyber.com/api/graph/query';
+  const modifier = {
+    period: 'all',
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
```

Comment:
What exactly do "left" and "right" mean in the lookingglass request? I wonder if we might want to briefly document it here so future Crossfeed developers can quickly get the context.

Reply:
We asked them, and they said it was a left and right join in their database, and they thought it would be a good idea to name it that, but it isn't very intuitive or good naming